src/drivers/ports/stm32f1/uart_port.h

```diff
@@ -31,7 +31,44 @@
 #ifndef __DRIVERS_UART_PORT_H__
 #define __DRIVERS_UART_PORT_H__
 
-#define UART_PORT_FRAME_FORMAT_DEFAULT 0
+/*
+ * STM32F1xx product line supports 8/9 data bits, N/E/O parity, 1/2 stop bits
+ * NOTE format field codification:
+ *      bits 0-7 are 8-15 of CR1 (coded 8 bits to the right)
+ *      bits 8-15 are 8-15 of CR2
+ */
+#define STM32_UART_DATA_BITS_8  0x00
+#define STM32_UART_DATA_BITS_9  ((0x01 << STM32_USART_CR1_M_POS) >> 8)
+#define STM32_UART_PARITY_N     0x00
+#define STM32_UART_PARITY_E     ((0x02 << STM32_USART_CR1_PS_POS) >> 8)
+#define STM32_UART_PARITY_O     ((0x03 << STM32_USART_CR1_PS_POS) >> 8)
+#define STM32_UART_STOP_BITS_1  0x00
+#define STM32_UART_STOP_BITS_2  (0x02 << STM32_USART_CR2_STOP_POS)
+#define UART_PORT_FRAME_FORMAT_8N1 (STM32_UART_DATA_BITS_8 \
```

Comment:
Well, CR2_STOP is a 2-bits field and it matches with single bit defines. Using _POS is the pattern I have followed in other MCUs defines. Anyway, the result is indifferent. Please, modify the code as you believe it is the right way 😉